.github/workflows/release.yml

```
@@ -33,3 +49,192 @@ jobs:
+          # Invariant: the release-workflow shape test asserts exactly 10
+          # asset paths in the `files:` block below. If this count is
+          # ever off, the matrix or `files:` list drifted from TARGETS.
+          expected=10
+          actual=$(find dist/release -maxdepth 1 -type f | wc -l)
+          if [[ "$actual" -ne "$expected" ]]; then
+            echo "::error::expected ${expected} release assets (5 binaries + 5 .sha512 sidecars), found ${actual}"
+            exit 1
+          fi
+
+      - name: Publish release assets
+        uses: softprops/action-gh-release@v1
```

actionlint (1.7.12)

[error] the runner of "softprops/action-gh-release@v1" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)